### PR TITLE
Coc Performance Measurement updates

### PR DESCRIPTION
**Pull request by eanders**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

This adds household type calculations to the summary metrics and fixes a bug where the comparison "year" was the same as the report year.

## Type of change
[//]: # 'remove options that are not relevant'
- [x] Bug fix

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 1]
```diff
@@ -303,7 +303,8 @@ def can_see_client_details?(user)
               parts[:questions].each do |question|
                 report_client["#{variant_name}_#{question[:name]}"] = question[:value_calculation].call(member)
                 hud_project_ids.each do |project_id|
-                  project_clients << {
+                  project_clients[hud_client.id] ||= Set.new
+                  project_clients[hud_client.id] << {
```
Comment (awisema) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb]: Maybe move this into a method so it doesn't have to get repeated every time? Not worth a class, but `add_client(project_clients, { ... })` maybe?

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 2]
```diff
@@ -366,14 +369,21 @@ def can_see_client_details?(user)
             client["#{variant_name}_#{parts[:key]}"] = value
             next unless value
 
+            # Use the previously calculated household_type, for now, just get the first for the client
+            # that matches one of the prior calculations
+            project_client = project_clients[client_id].detect do |pc|
```
Comment (awisema) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb]: Is this safe? Should it be
```suggestion
            project_client = project_clients[client_id]&.detect do |pc|
```

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 3]
```diff
@@ -366,14 +369,21 @@ def can_see_client_details?(user)
...
-            project_clients << {
+            project_clients[client_id] ||= Set.new
+            project_clients[client_id] << {
               report_id: id,
               client_id: client_id,
               project_id: nil,
               for_question: parts[:key], # allows limiting for a specific response
               period: variant_name,
-              household_type: nil,
+              household_type: project_client[:household_type],
```
Comment (awisema) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb]: Same question --  either 379 will never fire, or I think this needs to be
```suggestion
              household_type: project_client.try(:[], :household_type),
```
?

**Review by awisema — APPROVED** [review]

File: drivers/performance_measurement/app/models/performance_measurement/report.rb (hunk 2 above)
Comment (eanders) [reply drivers/performance_measurement/app/models/performance_measurement/report.rb]: Well, it _should_ be safe, but probably worth catching.

File: drivers/performance_measurement/app/models/performance_measurement/report.rb (hunk 1 above)
Comment (eanders) [reply drivers/performance_measurement/app/models/performance_measurement/report.rb]: Yeah, I had similar thoughts...  will do.

**Review by awisema — APPROVED** [review]
Looks good thanks!

**Pull request merged**